algo/src/main/java/org/neo4j/graphalgo/similarity/SimilarityProc.java

```diff
@@ -214,7 +214,22 @@ Double getSimilarityCutoff(ProcedureConfiguration configuration) {
         WeightedInput[] inputs = new WeightedInput[data.size()];
         int idx = 0;
         for (Map<String, Object> row : data) {
-            List<Number> weightList = (List<Number>) row.get("weights");
+
```

Comment:
I think this is ok, should also be fastest. We should probably pull this out into a util function.

An alternative (but slower) would be to check if the class of the type is an Array and the component-type is a subtype is number and iterate with reflection.

Reply:
We will also need it for Euclidean.

Reply:
Added a test for Euclidean and updated our Jaccard processing code too